flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/src/reader/BulkFormat.java

```diff
@@ -92,30 +91,22 @@
  * handed over as one.
  */
 @PublicEvolving
-public interface BulkFormat<T> extends Serializable, ResultTypeQueryable<T> {
+public interface BulkFormat<T, SplitT extends FileSourceSplit> extends Serializable, ResultTypeQueryable<T> {
 
 	/**
-	 * Creates a new reader that reads from {@code filePath} starting at {@code offset} and reads
-	 * until {@code length} bytes after the offset.
+	 * Creates a new reader that reads from the {@link FileSourceSplit#path() split's path}
+	 * starting at the {@link FileSourceSplit#offset()} split's offset} and reads
+	 * {@link FileSourceSplit#length() length} bytes after the offset.
 	 */
-	BulkFormat.Reader<T> createReader(
-			Configuration config,
-			Path filePath,
-			long splitOffset,
-			long splitLength) throws IOException;
+	BulkFormat.Reader<T> createReader(Configuration config, SplitT split) throws IOException;
```

Comment:
Looks like we can merge `createReader` and `restoreReader`, Format can be judged by `FileSourceSplit.getReaderPosition` like `FileSourceSplitReader.checkSplitOrStartNext`.

I don't have a strong tendency: A separate `restore` looks clearer. But may bring some `assert` like `StreamFormatAdapter` do.

Reply:
I was torn between the two options as well:
  - Have only one method, to allow consolidating/unifying the code
  - Have two methods, to make it explicit when restore happens. Less chance for users to not be aware of that restores need to be taken into account.

I decided for the second variant, because ensuring that users do not accidentally overlook the restore aspects seemed more important to me here.